src/canvas/Canvas.ts

```diff
@@ -1055,19 +1042,15 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
     this._isClick = true;
     this._cacheTransformEventData(e);
     this._handleEvent(e, 'down:before');
+
     let target: FabricObject | undefined = this._target;
-    // if right click just fire events
-    if (checkClick(e, RIGHT_CLICK)) {
-      if (this.fireRightClick) {
-        this._handleEvent(e, 'down', RIGHT_CLICK);
-      }
-      return;
-    }
 
-    if (checkClick(e, MIDDLE_CLICK)) {
-      if (this.fireMiddleClick) {
-        this._handleEvent(e, 'down', MIDDLE_CLICK);
-      }
+    // if right/middle click just fire events
+    if ((e as MouseEvent).button) {
+      this[FIRE_CLICK_EVENT_MAP[(e as MouseEvent).button as 1 | 2]] &&
```

Comment:
```suggestion
      if (this.fireMiddleClick && e.button === 1 || this.fireRightClick && e.button === 2) {
        .. handle event
        .. reset transform data
        return
      }
```

i find this way more readable, maybe is slight more code, but the map to reference a name of a property is so much indirect i can't read it.

this should also make not necessary the outer if for the e.button

I understand that we are not considering the buttons from 3 to 5, nor when multiple buttons are down, but that is a separate issue i m also not sure which is the correct way to handle it.

Reply:
will do

Reply:
done